big_tests/tests/cets_disco_SUITE.erl

```diff
@@ -67,44 +73,65 @@ rdbms_backend(_Config) ->
-    State2 = disco_init(Opts2),
-    {{ok, Nodes}, State2_2} = disco_get_nodes(State2),
+    State1 = disco_init(mim(), Opts1),
+    disco_get_nodes(mim(), State1),
+    State2 = disco_init(mim2(), Opts2),
+    {{ok, Nodes}, State2_2} = disco_get_nodes(mim2(), State2),
     %% "test2" node can see "test1"
     true = lists:member(test1, Nodes),
-    {{ok, _}, State2_3} = disco_get_nodes(State2_2),
+    {{ok, _}, State2_3} = disco_get_nodes(mim2(), State2_2),
     %% Check that we follow the right code branch
     #{last_query_info := #{already_registered := true}} = State2_3.
 
-rdbms_backend_supports_auto_cleaning(_Config) ->
+rdbms_backend_supports_auto_cleaning(Config) ->
+    ensure_mocked(Config),
     CN = <<"big_test2">>,
-    Opts1 = #{cluster_name => CN, node_name_to_insert => <<"test1">>, override_timestamp => month_ago()},
+    Opts1 = #{cluster_name => CN, node_name_to_insert => <<"test1">>},
     Opts2 = #{cluster_name => CN, node_name_to_insert => <<"test2">>},
-    %% test1 row is written
-    State1 = disco_init(Opts1),
-    {_, State1_2} = disco_get_nodes(State1),
-    {{ok, Nodes1}, _} = disco_get_nodes(State1_2),
+    %% test1 row is written with an old (mocked) timestamp
+    State1 = disco_init(mim(), Opts1),
+    {_, State1_2} = disco_get_nodes(mim(), State1),
+    {{ok, Nodes1}, State1_3} = disco_get_nodes(mim(), State1_2),
+    Timestamp = proplists:get_value(mocked_timestamp, Config),
+    #{last_query_info := #{timestamp := Timestamp}} = State1_3,
     %% It is in DB
     true = lists:member(test1, Nodes1),
     %% test2 would clean test1 registration
-    State2 = disco_init(Opts2),
-    {{ok, Nodes2}, State2_2} = disco_get_nodes(State2),
+    %% We don't mock on mim2 node, so timestamps would differ
+    State2 = disco_init(mim2(), Opts2),
+    {{ok, Nodes2}, State2_2} = disco_get_nodes(mim2(), State2),
     false = lists:member(test1, Nodes2),
     #{last_query_info := #{run_cleaning_result := {removed, [test1]}}} = State2_2.
 
 %%--------------------------------------------------------------------
 %% Helpers
 %%--------------------------------------------------------------------
 
-disco_init(Opts) ->
-    rpc(mim(), mongoose_cets_discovery_rdbms, init, [Opts]).
+disco_init(Node, Opts) ->
+    rpc(Node, mongoose_cets_discovery_rdbms, init, [Opts]).
 
-disco_get_nodes(State) ->
-    rpc(mim(), mongoose_cets_discovery_rdbms, get_nodes, [State]).
+disco_get_nodes(Node, State) ->
+    rpc(Node, mongoose_cets_discovery_rdbms, get_nodes, [State]).
 
 timestamp() ->
     os:system_time(second).
 
 month_ago() ->
     timestamp() - timer:hours(24 * 30) div 1000.
+
+mock_timestamp(Node, Timestamp) ->
+    ok = rpc(Node, meck, new, [mongoose_rdbms_timestamp, [passthrough, no_link]]),
+    ok = rpc(Node, meck, expect, [mongoose_rdbms_timestamp, select, 0, Timestamp]),
+    %% Ensure that we mock
+    EnsureMocked = fun() ->
+        Timestamp = rpc(Node, mongoose_rdbms_timestamp, select, [])
+        end,
+    EnsureMocked(),
```

Comment:
Execute and ignore result?

Reply:
I am matching Timestamp variable here.

Reply:
I.e. if meck fails to mock for some reason, we would get:

```
=== ERROR! init_per_testcase crashed!
	Location: [{cets_disco_SUITE,'-mock_timestamp/2-fun-0-',[127](file:///Users/mikhailuvarov/erlang/esl/MongooseIM/big_tests/ct_report/ct_run.test@MikhailU.2023-09-01_17.25.22/big_tests.tests.cets_disco_SUITE.logs/run.2023-09-01_17.25.23/cets_disco_suite.src.html#127)},
              {cets_disco_SUITE,mock_timestamp,[129](file:///Users/mikhailuvarov/erlang/esl/MongooseIM/big_tests/ct_report/ct_run.test@MikhailU.2023-09-01_17.25.22/big_tests.tests.cets_disco_SUITE.logs/run.2023-09-01_17.25.23/cets_disco_suite.src.html#129)},
              {cets_disco_SUITE,init_per_testcase,[50](file:///Users/mikhailuvarov/erlang/esl/MongooseIM/big_tests/ct_report/ct_run.test@MikhailU.2023-09-01_17.25.22/big_tests.tests.cets_disco_SUITE.logs/run.2023-09-01_17.25.23/cets_disco_suite.src.html#50)},
              {test_server,do_init_per_testcase,1552},
              {test_server,run_test_case_eval1,1253},
              {test_server,run_test_case_eval,1223}]
	Reason: {{badmatch,1693581924},
 [{cets_disco_SUITE,'-mock_timestamp/2-fun-0-',2,
                    [{file,"cets_disco_SUITE.erl"},{line,127}]},
  {cets_disco_SUITE,mock_timestamp,2,
                    [{file,"cets_disco_SUITE.erl"},{line,129}]},
  {cets_disco_SUITE,init_per_testcase,2,
                    [{file,"cets_disco_SUITE.erl"},{line,50}]},
  {test_server,do_init_per_testcase,2,[{file,"test_server.erl"},{line,1552}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1253}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,1223}]}]}
  ```